crates/ruff_linter/src/rules/flake8_simplify/rules/if_else_block_instead_of_if_exp.rs

```diff
@@ -111,6 +111,17 @@ pub(crate) fn if_else_block_instead_of_if_exp(checker: &mut Checker, stmt_if: &a
         return;
     }
 
+    let locator = checker.locator();
+
+    // If the original expression has a line break
+    // in either the "if-value" or the "else-value",
+    // don't suggest to turn it into a ternary expression
+    if locator.contains_line_break(body_value.range())
+        || locator.contains_line_break(else_value.range())
+    {
+        return;
+    }
```

Comment:
This restriction wasn't necessary with the old approach that used the expression generator, but it makes the "cut-and-paste" approach much easier. It makes a lot of existing violations in the ecosystem check go away.

However, I think that might actually be desirable? Here's an `if`/`else` expression in `cibuildwheel` that's no longer getting flagged. I think it's pretty clear from looking at it that they've deliberately chosen an `if`/`else` block here instead of a ternary precisely _because_ the expression in the `if` block stretches over multiple lines. It wouldn't really simplify the code here to change this into a ternary expression:

```py
if args.schemastore:
    non_global_options = {
        k: {"$ref": f"#/properties/tool/properties/cibuildwheel/properties/{k}"}
        for k in schema["properties"]
    }
else:
    non_global_options = {k: {"$ref": f"#/properties/{k}"} for k in schema["properties"]}
```

So arguably this new restriction gets rid of a bunch of false positives :)

Reply:
I generally agree with the change. Only question for me is whether it should be in preview. I would be comfortable shipping it to stable since it's a _reduction_ in scope, so shouldn't generally introduce new violations. \cc @zanieb for a second opinion.

Reply:
Yeah... I guess the only "backwards-incompatible" change here is that people might have `# noqa` comments in their code to suppress violations on this kind of code — those comments could now be flagged by ruff as unused? But I feel like people will mostly be happy if they get to delete `# noqa` comments, so I lean towards arguing it should go straight into stable. But no strong opinion either way :)

Reply:
Yeah I feel the same.

Reply:
Oh interesting... I prefer to write these as ternary expressions even when there are line breaks and would be surprised to see the lint go away. I don't think there's any requirement that we ship a reduction in scope in preview but I think we may want to here. I don't actually see any "unused noqa" changes in the ecosystem so I don't know if we have strong evidence that a line break is a common reason for ignoring this rule. 

A search for [`# noqa: SIM108`](https://github.com/search?q=%23+noqa%3A+SIM108&type=code) may be interesting. Very few cases where there are multiple lines in the brief perusal I just did. It seems like we might want to add an exception to `if TYPE_CHECKING` blocks though!